### PR TITLE
Migrate gometalinter to golangci lint

### PR DESCRIPTION
**Pull request by joonas**
**What this PR does / why we need it**:

[`gometalinter`](https://github.com/alecthomas/gometalinter) has been deprecated, and [`golangci-lint`](https://github.com/golangci/golangci-lint) is the recommended replacement.

This replaces `gometalinter` with `golangci-lint`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+run:
+  concurrency: 2
```
Comment (xmudrii) [inline .golangci.yml]: What is the default behavior, i.e. when we don't specify concurrency?

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+linters:
+  disable-all: true
+  enable:
+    - vet
+    - golint
+    - varcheck
+    - structcheck
+    - gosimple
+    - unused
+    - errcheck
```
Comment (xmudrii) [inline .golangci.yml]: Looking at other CAPI projects, they enable a bit more linters. Is it possible to enable those here too?

File: hack/tools/go.mod [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module sigs.k8s.io/cluster-api-provider-digitalocean/hack/tools
```
Comment (xmudrii) [inline hack/tools/go.mod]: This seems to be in the scope of #137. Is this PR supposed to supersede #137?

**Comment by cpanato** [comment]
this PR is still active? what we can do to move forward? if needed i can commit the requested changes.
let me know

@joonas @xmudrii

**Comment by xmudrii** [comment]
@cpanato @joonas Seems like #140 is also based on gometalinter, so maybe we can even close this PR if that sounds okay.

**Comment by xmudrii** [comment]
Closing this PR in favor of #140.

**Pull request closed without merging**